pkg/asset/installconfig/kubevirt/client.go

```diff
@@ -0,0 +1,213 @@
+package kubevirt
+
+import (
+	"context"
+	"io/ioutil"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	unstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	kubevirtapiv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+//go:generate mockgen -source=./client.go -destination=./mock/client_generated.go -package=mock
+
+// Client is a wrapper object for actual infra-cluster clients: kubernetes and the kubevirt
+type Client interface {
+	GetVirtualMachine(ctx context.Context, namespace string, name string) (*kubevirtapiv1.VirtualMachine, error)
+	ListVirtualMachine(ctx context.Context, namespace string, opts metav1.ListOptions) (*kubevirtapiv1.VirtualMachineList, error)
+	DeleteVirtualMachine(ctx context.Context, namespace string, name string) error
+	GetDataVolume(ctx context.Context, namespace string, name string) (*cdiv1.DataVolume, error)
+	ListDataVolume(ctx context.Context, namespace string, opts metav1.ListOptions) (*cdiv1.DataVolumeList, error)
+	DeleteDataVolume(ctx context.Context, namespace string, name string) error
+	GetSecret(ctx context.Context, namespace string, name string) (*corev1.Secret, error)
+	ListSecret(ctx context.Context, namespace string, opts metav1.ListOptions) (*corev1.SecretList, error)
+	DeleteSecret(ctx context.Context, namespace string, name string) error
+	GetStorageClass(ctx context.Context, name string) (*storagev1.StorageClass, error)
+	GetNetworkAttachmentDefinition(ctx context.Context, name string, namespace string) (*unstructured.Unstructured, error)
+}
+
+type client struct {
+	kubernetesClient *kubernetes.Clientset
+	dynamicClient    dynamic.Interface
+}
+
+var (
+	vmRes = schema.GroupVersionResource{
+		Group:    kubevirtapiv1.GroupVersion.Group,
+		Version:  kubevirtapiv1.GroupVersion.Version,
+		Resource: "virtualmachines",
+	}
+
+	dvRes = schema.GroupVersionResource{
+		Group:    cdiv1.SchemeGroupVersion.Group,
+		Version:  cdiv1.SchemeGroupVersion.Version,
+		Resource: "datavolumes",
+	}
+)
+
+// LoadKubeConfigContent returns the kubeconfig file content
+func LoadKubeConfigContent() ([]byte, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeConfigFilename := loadingRules.GetDefaultFilename()
+
+	return ioutil.ReadFile(kubeConfigFilename)
```

Comment:
Let's stay consistent between `NewClient` and this function so that there are not any surprises.

I also think that we want to trim down the config to remove anything that is not related to the current context. We do not want to send the information for **all** of the clusters that the user may have in their kubeconfig. But we can do that work later.

```suggestion
	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
	configOverrides := &clientcmd.ConfigOverrides{}
	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
	return clientConfig.RawConfig().Write()
```

Reply:
Thank you @staebler
I open a task to change it later (OCPCNV-158)
Can we go with it as is now?